src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -10924,10 +10937,22 @@ HCIMPL2(EXTERN_C void, ProfileEnter, UINT_PTR clientData, void * platformSpecifi
 }
 HCIMPLEND
 
-HCIMPL2(EXTERN_C void, ProfileLeave, UINT_PTR clientData, void * platformSpecificHandle)
+HCIMPL2_RAW(EXTERN_C void, ProfileLeave, UINT_PTR clientData, void * platformSpecificHandle)
```

Comment:
Is the change actually needed in this helper? I would expect that the thread is always going to be setup by the time we setup the frame.

Reply:
Ah, I guess you are right that with the added thread struct setup in the ProfileEnter, the ProfileLeave can be left unchanged.

Reply:
w.r.t. the thread struct creation.